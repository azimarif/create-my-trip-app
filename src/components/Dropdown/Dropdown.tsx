import React from 'react';
import {
    FormControl,
    InputLabel,
    MenuItem,
    OutlinedInput,
    Select,
} from "@mui/material";

interface DropdownProps {
    items: string[];
    defaultValue: string;
    onChangeHandler: any
}

const Dropdown: React.FC<DropdownProps> = ({ items, defaultValue, onChangeHandler }) => {
    return (
        <FormControl sx={{ m: 0, width: 300 }}>
            <InputLabel>{defaultValue}</InputLabel>
            <Select
                onChange={onChangeHandler}
                input={<OutlinedInput label={defaultValue} />}
            >
                {items.map((item: string) => (
                    <MenuItem key={item} value={item}>
                        {item}
                    </MenuItem>
                ))}
            </Select>
        </FormControl>
    );
}

export default Dropdown;
