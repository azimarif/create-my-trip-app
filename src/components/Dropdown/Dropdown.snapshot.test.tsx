import React from 'react';
import { render } from '@testing-library/react';
import Dropdown from './Dropdown';

test('Dropdown component matches snapshot', () => {
  const items = ['Option 1', 'Option 2', 'Option 3'];
  const defaultValue = 'Select an option';

  const { container } = render(
    <Dropdown
      items={items}
      defaultValue={defaultValue}
      onChangeHandler={jest.fn()}
    />,
  );

  expect(container).toMatchSnapshot();
});
