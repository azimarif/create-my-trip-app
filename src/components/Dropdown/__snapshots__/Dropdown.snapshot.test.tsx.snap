// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown component matches snapshot 1`] = `
<div>
  <div
    class="MuiFormControl-root css-cquceo-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
    >
      Select an option
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <div
        aria-controls=":r0:"
        aria-expanded="false"
        aria-haspopup="listbox"
        class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        role="combobox"
        tabindex="0"
      >
        <span
          class="notranslate"
        >
          â€‹
        </span>
      </div>
      <input
        aria-hidden="true"
        aria-invalid="false"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        tabindex="-1"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-yjsfm1"
        >
          <span>
            Select an option
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
